app/src/main/java/com/github/geohunt/app/ui/screens/teamprogress/TeamProgressScreen.kt

```diff
@@ -0,0 +1,76 @@
+package com.github.geohunt.app.ui.screens.teamprogress
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.geohunt.app.model.Challenge
+import com.github.geohunt.app.ui.screens.teamprogress.TeamProgressViewModel.TeamStatus
+import com.github.geohunt.app.ui.utils.pagination.FinitePagedList
+
+/**
+ * Creates the team progress screen.
+ */
+@Composable
+fun TeamProgressScreen(
+    onBack: () -> Unit,
+    onLeaderboard: () -> Unit,
+    onChat: () -> Unit,
+    onClaim: (Challenge) -> Unit,
+    bountyId: String,
+    viewModel: TeamProgressViewModel = viewModel(factory = TeamProgressViewModel.getFactory(bountyId))
+) {
+    val teamStatus = viewModel.teamStatus.collectAsStateWithLifecycle()
+
+    when(teamStatus.value) {
+        TeamStatus.LOADING_TEAM -> EmptyTeamProgressScreen(
+            title = "Loading...",
+            onBack = onBack
+        ) {
+            CircularProgressIndicator()
+        }
+
+        TeamStatus.ERROR_NO_TEAM -> EmptyTeamProgressScreen(
+            title = "Error",
+            onBack = onBack
+        ) {
+            Box(modifier = Modifier
+                .padding(it)
+                .fillMaxSize()) {
+                Text(
+                    "It seems that you are not registered in a team yet. " +
```

Comment:
String resources?

Reply:
Oh yes, forgot some. I'll add them in the polishing week.